### ISSUE TITLE
Rule request: no "self = this;"

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [palantir/tslint#2861]
A lot of older JS programs used to put `var self = this;` variables all over the place. We no longer need to do that now that arrow lambdas are available. Still using this is likely a symptom of not having converted over yet and/or not using captured scopes correctly.

Bad:

```typescript
private act() {
    var self = this; // unnecessary
    setTimeout(
        () => {
            self.complete();
        },
        this.delay);
}
```

Better:

```typescript
private act() {
    setTimeout(
        () => {
            this.complete();
        },
        this.delay);
}
```

This is in [tslint-microsoft-contrib as `no-var-self`](https://github.com/Microsoft/tslint-microsoft-contrib/blob/master/src/noVarSelfRule.ts), though a better name might be `no-self-equals-this`.

**Comment by reduckted** [comment]
Might be worth mentioning this in #1142.

**Comment by JoshuaKGoldberg** [comment]
:+1:. 

Some options for this rule that could be added in:
A. Names of allowed "self" equivalents (in the tslint-microsoft-contrib version)
B. Allow the variable if it's used in a later scope with a potentially different `this`.
